### ISSUE TITLE
Unable to remount after cleaning

### ISSUE THREAD

**Issue by rclabough** [LinearTapeFileSystem/ltfs#387]
**Describe the bug**
LTFS is unable to continue an operation after my drive has reported that it needs cleaning.  I've read the manuals, and I believe I understand why, the drive reports that it needs cleaning and stops writing.  I clean the drive with my cleaning cartridge, but on mounting again my cartridge has entered into a "write-perm" and "Volume Lock State = 0x04", and I cannot repair it.   I've attempted `ltfsck /dev/sg2 -z` to run a repair, but it won't do anything because the cartridge is locked.  This is my second attempt now, it spends 24 hours writing 6TB of data but then it needs a cleaning, but then I can never use the rest of the cartridge.  This happens on multiple cartridges.

**To Reproduce**
Run a long write operation, spanning at least a day, drive gets dirty.
Drive enters into the failsafe "write perm mode", but there's no option to run the repair tools as the drive is locked.

**Expected behavior**
As far as I can tell from the manuals, `ltfsck` is supposed to be able to repair this if the user is confirming that the cartridge is still okay.  From what I see there is no way to unlock the cartridge without reformatting and starting everything over, leading to a very frustrating cycle.

**Screenshots**
```
$ sudo ltfsck /dev/sg2 -z
LTFS16000I Starting ltfsck, LTFS version 2.5.0.0 (Prelim), log level 2.
LTFS16088I Launched by "ltfsck -z /dev/sg2".
LTFS16089I This binary is built for Linux (x86_64).
LTFS16090I GCC version is 11.3.0.
LTFS17087I Kernel version: Linux version 5.19.0-35-generic (buildd@lcy02-amd64-020) (x86_64-linux-gnu-gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0, GNU ld (GNU Binutils for Ubuntu) 2.38) #36~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Fri Feb 17 15:17:25 UTC 2 i386.
LTFS17089I Distribution: PRETTY_NAME="Ubuntu 22.04.2 LTS".
LTFS17089I Distribution: DISTRIB_ID=Ubuntu.
LTFS17085I Plugin: Loading "sg" tape backend.
LTFS30209I Opening a device through sg-ibmtape driver (/dev/sg2).
LTFS30250I Opened the SCSI tape device 6.0.0.0 (/dev/sg2).
LTFS30207I Vendor ID is IBM     .
LTFS30208I Product ID is ULTRIUM-HH8     .
LTFS30214I Firmware revision is J4D1.
LTFS30215I Drive serial is 10WT011570.
LTFS30285I The reserved buffer size of /dev/sg2 is 1048576.
LTFS30205I RSOC (0xa3) returns -20601.
LTFS30263I RSOC returns Not Ready to Ready Transition, Medium May Have Changed (-20601) /dev/sg2.
LTFS30262I Forcing drive dump.
LTFS39802W Unknown SCSI OP code 0x1d, use default timeout.
LTFS30253I Saving drive dump to /tmp/ltfs_10WT011570_2023_0324_222038_f.dmp.
LTFS39802W Unknown SCSI OP code 0x3c, use default timeout.
LTFS39802W Unknown SCSI OP code 0x3c, use default timeout.
LTFS39802W Unknown SCSI OP code 0x3c, use default timeout.
LTFS39802W Unknown SCSI OP code 0x3c, use default timeout.
LTFS39802W Unknown SCSI OP code 0x3c, use default timeout.
LTFS39802W Unknown SCSI OP code 0x3c, use default timeout.
LTFS39802W Unknown SCSI OP code 0x3c, use default timeout.
LTFS39802W Unknown SCSI OP code 0x3c, use default timeout.
LTFS39802W Unknown SCSI OP code 0x3c, use default timeout.
LTFS39802W Unknown SCSI OP code 0x3c, use default timeout.
LTFS39802W Unknown SCSI OP code 0x3c, use default timeout.
LTFS39802W Unknown SCSI OP code 0x3c, use default timeout.
LTFS39802W Unknown SCSI OP code 0x3c, use default timeout.
LTFS39802W Unknown SCSI OP code 0x3c, use default timeout.
LTFS39802W Unknown SCSI OP code 0x3c, use default timeout.
LTFS30294I Setting up timeout values from vendor and device.
LTFS17160I Maximum device block size is 1048576.
LTFS11330I Loading cartridge.
LTFS30252I Logical block protection is disabled.
LTFS11332I Load successful.
LTFS17157I Changing the drive setting to write-anywhere mode.
LTFS16014I Checking LTFS file system on '/dev/sg2'.
LTFS30252I Logical block protection is disabled.
LTFS16023I LTFS volume information:.
LTFS16024I Volser (bar code) :       .
LTFS16025I Volume UUID     : 47403d8a-b905-4cbd-8df3-7f45d39d125e.
LTFS16026I Format time     : 2023-03-23 22:59:01.371667246 PDT.
LTFS16027I Block size      : 524288.
LTFS16028I Compression     : Enabled.
LTFS16029I Index partition : ID = a, SCSI Partition = 0.
LTFS16030I Data partition  : ID = b, SCSI Partition = 1.

LTFS16111I The recovery process is skipped because of a locked cartridge (4).
LTFS11005I Mounting the volume from device.
LTFS30252I Logical block protection is disabled.
LTFS11333I A cartridge with write-perm error is detected on DP. Seek the newest index (IP: Gen = 173, VCR = 1744) (DP: Gen = 172, VCR = 1709) (VCR = 1744).
LTFS17286I VCR value is matched between IP-MAM and VCR (1744).
LTFS17287I Making R/O mount from the location (a, 8).
LTFS17227I Tape attribute: Vendor = IBM     .
LTFS17227I Tape attribute: Application Name = LTFS                            .
LTFS17227I Tape attribute: Application Version = 2.5.0.0 .
LTFS17227I Tape attribute: Medium Label = .
LTFS17228I Tape attribute: Text Localization ID = 0x81.
LTFS17227I Tape attribute: Barcode =                                 .
LTFS17227I Tape attribute: Application Format Version = 2.4.0           .
LTFS17228I Tape attribute: Volume Lock Status = 0x04.
LTFS17227I Tape attribute: Media Pool name = .
LTFS11031I Volume mounted successfully. NO_BARCODE : Gen = 173 / (a, 8) -> (b, 8558048) / 10WT011570.
LTFS17265I Skip writing the index because of the volume is mounted as ERROR_TAPE.
LTFS11034I Volume unmounted successfully.
LTFS16022I Volume is consistent.
LTFS30252I Logical block protection is disabled.
```

I know these exist, and they might be what I'm looking for, but maybe I'm looking more for confirmation - what does `force_mount_no_eod` do exactly, is it what I'm looking for so I can just plow forward and continue copying? 

Should I in the future be mounting it differently? I see `sync_type`, and is there a way to say "write the EOD on error" or have some "correctable" error handling?

Sorry, I'm just very frustrated after a few days and desperately want my backups to work.  Any help is very appreciated

**Comment by rclabough** [comment]
To be clear, this is for personal use, I do not care about the bits that may have been lost.  I just want to be able to continue my copy job to fill the rest of the cartridge, or know what I'm doing wrong because it feels like it's impossible to do a big copy job without something failing halfway.

**Comment by piste-jp-ibm** [comment]
This behavior is as designed.

LTFS makes a tape `READ ONLY` when it receives a permanent write error (because of medium reason) from the drive. That record is stored into the MAM and cannot be cleared except reformat. (See the section, `10.5 Volume Advisory Locking`, in the format spec.)

That's because tape device is sequential access device. In other words, tape device always must append new data at the end of data stream even if it is update of file. So permanent write error because of medium reason (like medium defect) is really critical.

If a permanent write error because of medium defect, following condition might happen. So we strongly recommend that user copy the data on tape to another tape (or disk) and dispose write permed tape. It's the reason that LTFS only allows to read the data on tape.

1. User might get another permanent write error at the same position even if user recovers the tape by `ltfsck`
2. User might get another permanent write error at the same position again even if user reformat the tape

See the following section in the IBM docs also.

- https://www.ibm.com/docs/en/spectrum-archive-sde/2.4.5?topic=troubleshooting-how-write-errors-are-handled

In IBM Spectrum Archive LE, tape in this condition is reported as `DEGRADED(WRITE_FENCED)`, you can see the next action we recommend.

- https://www.ibm.com/docs/en/spectrum-archive-le/2.4.5?topic=tape-status-codes#ltfs_tape_status_codes

**Comment by rclabough** [comment]
hi, to come full circle to this I found that this had already been brought up and solved.  It took a lot of digging, but my drive is an ultrium drive and it looks like it fell under the `enable-buggy-ifs` category.  This was driving me bonkers because everything I tried seem to fail, but after recompiling with option (and also doing a firmware upgrade) I have finally been able to complete a full LTO-8 tape!

For others, read the man pages, but if you have an ultrium drive and it seems that it's kicking out every tape for write errors and asking to be cleaned all the time, try 1) Upgrading your firmware and 2) Compile LTFS with the [enable-buggy-ifs](https://github.com/LinearTapeFileSystem/ltfs/wiki/HBA-info#note-for-buggy-hbas) switch.  I found it deep in a forum that Ultrium drives while not listed in this wiki, have the same problem.
